nautobot/dcim/models/device_component_templates.py

```diff
@@ -355,6 +356,9 @@ class InterfaceTemplate(ModularComponentTemplateModel):
         blank=True,
     )
     type = models.CharField(max_length=50, choices=InterfaceTypeChoices)
+    port_type = models.CharField(
+        max_length=50, choices=PortTypeChoices, blank=True, help_text="Physical connector type"
```

Comment:
Might be nice to enhance `nautobot.dcim.factory.InterfaceTemplateFactory` to sometimes create interface-templates with a non-blank `port_type` for test purposes, though it'd have to be smart enough not to run afoul of the added clean logic?

(For that matter, I realize we missed adding `speed` and `duplex` to the factory recently - ah well :smile:)